.github/workflows/CI-docker.yml

```diff
@@ -1,45 +1,30 @@
 name: CI-docker
-run-name: "Workflow performing CI step: push docker image to DockerHub"
+run-name: "Workflow performing CI step: Docker Image Build and Publish"
 
 on:
   push:
     tags:
-      - v[0-9].[0-9].[0-9]*
+      - '[0-9].[0-9].[0-9]*'
   workflow_dispatch:
 
 env:
   RUST_BACKTRACE: 1
+  PROD_RELEASE_BRANCH: main
 
 jobs:
   docker:
-    strategy:
-      matrix:
-        os: [ubuntu-latest]
-        # os: [ubuntu-latest, windows-latest, macos-latest]
-    runs-on: ${{ matrix.os }}
+    runs-on: ubuntu-latest
     steps:
-      - name: Checkout branch
+      - name: Git checkout
         uses: actions/checkout@v4
-      - name: Build Docker image
-        run: |
-          docker build \
-          --build-arg VCS_REF="${VCS_REF}" \
-          --build-arg BUILD_DATE="$(date -u '+%Y-%m-%dT%H:%M:%SZ')" \
-          --build-arg IMAGE_NAME="hl/nh-node" \
-          -f docker/dockerfiles/hl-node.Dockerfile -t nh-node:build .
-      - name: Login into DockerHub
-        run: docker login --username cihorizen --password ${{ secrets.DOCKER_HUB_TOKEN }}
-      - name: Publish Docker image(s) to DockerHub
+        with:
```

Comment:
I see no need for this:
```
        with:
          ref: ${{ github.ref }}
```

Reply:
This is actually a very important one to have. The pipeline will not work correctly without it. Took me couple hours to figure this out :face_exhaling: . Long story shout without this `ref` Github Tag will be checked out by its commit hash vs tag therefore `git verify-tag` command that checks if git tag is signed by the maintainers_key will fail since it ONLY works on the object = tag and not object = commit.

We should keep this in mind for the future cases like this.